### PR TITLE
Remove render modifiers in school components

### PR DESCRIPTION
**Pull request by jrjohnson**
I did these all at once as I was working on some data loading at the same time, just going to push this PR first though as it's big enough already. Each component in its own commit, but still a big change. Sorry.

### CODE REVIEW

File: packages/frontend/app/components/school-competencies-expanded.js [hunk 1]
```diff
@@ -1,27 +1,27 @@
...
-import { dropTask, restartableTask } from 'ember-concurrency';
+import { dropTask } from 'ember-concurrency';
 import { filterBy, uniqueValues } from 'ilios-common/utils/array-helpers';
+import { TrackedAsyncData } from 'ember-async-data';
 
 export default class SchoolCompetenciesExpandedComponent extends Component {
   @service store;
   @tracked competenciesToAdd = [];
   @tracked competenciesToRemove = [];
-  @tracked schoolCompetencies;
```
Comment (stopfstedt) [inline packages/frontend/app/components/school-competencies-expanded.js:13]: needs to be replaced with a getter that wraps around `get schoolCompetenciesData()`.

**Review by stopfstedt — COMMENTED** [review]
line:  `this.schoolCompetencies = await this.args.school.competencies;` in `SchoolCompetenciesExpandedComponent::save()` should be removed.

File: packages/frontend/app/components/school-vocabularies-expanded.js [hunk 2]
```diff
@@ -1,14 +1,13 @@
...
 import { service } from '@ember/service';
-import { dropTask, restartableTask } from 'ember-concurrency';
-import { map } from 'rsvp';
+import { dropTask } from 'ember-concurrency';
 import { findById } from 'ilios-common/utils/array-helpers';
+import { TrackedAsyncData } from 'ember-async-data';
 
 export default class SchoolVocabulariesExpandedComponent extends Component {
   @service store;
-  @tracked schoolVocabularies = [];
```
Comment (stopfstedt) [inline packages/frontend/app/components/school-vocabularies-expanded.js:11]: something broke in this component refactor. when adding and saving a new vocabulary, i get this error coming out of this component in the browser console `Uncaught (in promise) Error: Assertion Failed: Expected id to be a string or number, received null`

File: packages/frontend/app/components/school-session-type-form.hbs [hunk 3]
```diff
@@ -2,13 +2,10 @@
   <div
     class="school-session-type-form"
     data-test-school-session-type-form
-    {{did-insert (perform this.load)}}
```
Comment (stopfstedt) [inline packages/frontend/app/components/school-session-type-form.hbs:5]: something broken with this refactor. clicking the "save" button on this form does nothing now now - regardless of updating or adding an item.

**Review by stopfstedt — CHANGES_REQUESTED** [review]
several components turned out to be broken during click-testing. might be a good opportunity to add acceptance tests while at, we are very light on it right now.

**Comment by jrjohnson** [comment]
Thanks for saving me from myself! Clearing was overconfident in our test coverage. I'll add a bunch of tests.

File: packages/frontend/app/components/school-vocabularies-expanded.js (hunk 2 above)
Comment (jrjohnson) [reply packages/frontend/app/components/school-vocabularies-expanded.js:11]: ✅

File: packages/frontend/app/components/school-manager.js [hunk 4]
```diff
@@ -11,8 +11,8 @@ export default class SchoolManagerComponent extends Component {
   @service flashMessages;
   @tracked @NotBlank() @Length(1, 60) title;
 
-  @action
-  load() {
+  constructor() {
```
Comment (stopfstedt) [inline packages/frontend/app/components/school-manager.js:14]: :D

**Pull request merged**